### PR TITLE
Delete Secrets when Machine is Deleted

### PR DESCRIPTION
**Pull request by AbdullahAlShaad**

### CODE REVIEW

**Review by Superm4n97 — DISMISSED** [review]
LGTM!

**Review by Superm4n97 — DISMISSED** [review]
LGTM!

File: api/v1alpha1/helper.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func GetPhase(obj *Machine) MachinePhase {
```
Comment (masudur-rahman) [inline api/v1alpha1/helper.go:87]: Don't panic (line: 80)

Comment (AbdullahAlShaad) [reply api/v1alpha1/helper.go:87]: Updated

**Review by Superm4n97 — APPROVED** [review]
LGTM!

**Pull request merged**